### PR TITLE
Add preprocessor extract_point function

### PR DESCRIPTION
**Pull request by evertrol**
This add a simple point extraction function the preprocessor, with interpolation. It uses `iris.cube.Cube.interpolate` under the hood.

Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValCore/blob/master/CONTRIBUTING.md).

**Tasks**

-   [ ] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [x] Add unit tests
-   [x] Public functions should have a numpy-style docstring so they appear properly in the [API documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/api/esmvalcore.html). For all other functions a one line docstring is sufficient.
-   [x] If writing a new/modified preprocessor function, please update the [documentation](https://esmvaltool.readthedocs.io/projects/esmvalcore/en/latest/esmvalcore/preprocessor.html)
-   [ ] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [ ] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.

If you need help with any of the tasks above, please do not hesitate to ask by commenting in the issue or pull request.

* * *

Closes {Link to corresponding issue}

### CODE REVIEW

**Comment by evertrol** [comment]
Note that the last commit, 083bd78f, can be removed again, if so wanted: I just noticed some trailing whitespace and spelling mistakes when adding the function to the documentation, hence the fixes in the entire `preprocessor.rst` section. It's not really part of the actual functionality, though it felt too little to warrant its own PR (perhaps then, all of the documentation should be cleaned up in a separate PR). 
And I didn't bother with British versus US English!

File: doc/esmvalcore/preprocessor.rst [hunk 1]
```diff
@@ -986,10 +990,25 @@ Examples:
...
+
+Extract a single point from the data. This is done using either
+nearest or linear interpolation.
+
+Parameters:
+  * ``cube``: the input dataset cube.
+  * ``latitude``, ``longitude``: coordinates (as floating point
+    values) of the point to be extracted.
+  * ``scheme``: interpolation scheme: either ``'linear'`` or
+    ``'nearest'``. There is no default.
```
[the remark this answers is not in the document]
Comment (evertrol) [reply doc/esmvalcore/preprocessor.rst]: (Apologies for the late reply; I thought I had already replied, but never actually hit the "Comment" button.)

I thought about a default, but I wanted to make the choice explicit for the end-user.
I can imagine a situation where the result between the two choices could be quite different: if the relevant cell(s) are large, and the point chosen is relatively close to the edge of a cell, far away from the centre, then `'nearest'` would likely yield quite a different result than `'linear'`. Further, I wouldn't know what a good (best) default value would be.
I also wanted it to make it explicit that there is no option like 'area_weighted' (or latitude-weighted), for latitude interpolation. In contrast, `extract_point`s close neighbour, `regrid`, does have that option.
Another comparison with `regrid` shows that `regrid` also doesn't have a default interpolation/regridding schemes, so I've kept it the same for that reason as well.

**Comment by evertrol** [comment]
@bouweandela Will put in some further unit tests. The current one is just a limited version of the regrid one, which is completely mocked, but here, actual values are probably fine without taking too much.
That combines nicely with the suggestion to allow for arrays of latitude and/or longitude, which can be implemented together with an extended unit test.

**Comment by evertrol** [comment]
I've updated the PR with unit tests (I've tried to cover all cases; I hope I haven't gone overboard with the tests).
I noticed there are 'unit' and 'integration' tests. I followed the `regrid` function, so I've put these in 'integration', and left the existing one (with mock tests) in 'unit'.

I've also amended the docstring and documentation to mention latitude and longitude can be arrays, instead of scalars.

Question: is there a convention for the order of latitude, longitude?

Also, if someone knows how to skip the doctests for a complete docstring, that would be convenient: currently, there's a # doctest: +SKIP directive after each line, which just adds noise to the examples.

**Comment by bouweandela** [comment]
> Question: is there a convention for the order of latitude, longitude?

I think it's latitude longitude in all CMIP data.

File: doc/esmvalcore/preprocessor.rst (hunk 1 above)
[the remark this answers is not in the document]
Comment (valeriupredoi) [reply doc/esmvalcore/preprocessor.rst]: fair do's, cheers :beer:

**Comment by mattiarighi** [comment]
Thanks for this nice addition and for revising the documentation. 👍

**Pull request merged**